pkg/vault/test/foundry/RouterQueriesDiffRates.t.sol

```diff
@@ -0,0 +1,502 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.24;
+
+import "forge-std/Test.sol";
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { IBasePool } from "@balancer-labs/v3-interfaces/contracts/vault/IBasePool.sol";
+import { IRateProvider } from "@balancer-labs/v3-interfaces/contracts/vault/IRateProvider.sol";
+import { TokenConfig } from "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+
+import { ArrayHelpers } from "@balancer-labs/v3-solidity-utils/contracts/helpers/ArrayHelpers.sol";
+import { BasePoolMath } from "@balancer-labs/v3-solidity-utils/contracts/math/BasePoolMath.sol";
+import { FixedPoint } from "@balancer-labs/v3-solidity-utils/contracts/math/FixedPoint.sol";
+
+import { BaseVaultTest } from "./utils/BaseVaultTest.sol";
+import { RateProviderMock } from "../../contracts/test/RateProviderMock.sol";
+
+contract RouterQueriesDiffRatesTest is BaseVaultTest {
+    using ArrayHelpers for *;
+    using FixedPoint for uint256;
+
+    // Track the indices for the standard dai/usdc pool.
+    uint256 internal daiIdx;
+    uint256 internal usdcIdx;
+
+    // A bigger pool init amount is needed, so we can manipulate the token rates safely, without breaking the linear
+    // math of PoolMock (Linear math can return amounts out outside of pool balances since it does not have protections
+    // in the edge of the pricing curve).
+    uint256 internal constant biggerPoolInitAmount = 1e6 * 1e18;
+
+    IRateProvider[] internal rateProviders;
+
+    function setUp() public override {
+        (daiIdx, usdcIdx) = getSortedIndexes(address(dai), address(usdc));
```

Comment:
How does this work? Doesn't it have to be after `setUp`, since that's where dai is deployed?

Reply:
Fixed!